src/redisearch_rs/rlookup/src/lookup.rs

```diff
@@ -1108,7 +1160,11 @@ impl<'a> RLookup<'a> {
             .keys
             .find_by_name_mut(name)
             .expect("key should have been created above");
-        let key = if let Some(fs) = self.index_spec_cache.as_ref()?.find_field(name) {
+        let key = if let Some(fs) = self
```

Comment:
This change actually fixes a bug that we haven't caught yet. The C `findFieldInSpecCache` treats "no spcache" the same as "no such field in spcache". This early return (`?`) in the Rust version wasn't correct. 

I added a test further below to assert this correct behavior going forward.